### PR TITLE
PLT-465: Added support for Force delete for all cloud clusters

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

**Comment by SivaanandM** [comment]
> As discussed let's have a separate timeout parameter.

@nikchern we can introduce separate attributes like force_delete_delay, but again we need to add validation minimum time is 15 as in UI. So, if force delete is enforced to cluster, then we should always maintain some constant time and do force delete.

**Comment by SivaanandM** [comment]
> As discussed let's have a separate timeout parameter.

done

File: spectrocloud/resource_cluster_libvirt.go [hunk 1]
```diff
@@ -303,6 +303,18 @@ func resourceClusterLibvirt() *schema.Resource {
...
+			"force_delete": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "If set to `true`, the cluster will be force deleted and user has to manually clean up the provisioned cloud resources.",
+			},
+			"force_delete_delay": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     60,
```
Comment (nikchern) [inline spectrocloud/resource_cluster_libvirt.go:315]:
```suggestion
				Default:     20,
```

File: spectrocloud/cluster_common_crud.go [hunk 2]
```diff
@@ -196,15 +196,24 @@ func resourceClusterDelete(ctx context.Context, d *schema.ResourceData, m interf
...
 	var diags diag.Diagnostics
-
-	err := c.DeleteCluster(d.Id())
+	var err error
+	err = c.DeleteCluster(d.Id())
 	if err != nil {
 		return diag.FromErr(err)
 	}
-
+	if forceDelete, ok := d.GetOk("force_delete"); ok && forceDelete == true {
```
[the remark this answers is not in the document]
Comment (SivaanandM) [reply spectrocloud/cluster_common_crud.go:204]: added validation in schema level

**Review by nikchern — APPROVED** [review]

**Pull request merged**